### PR TITLE
Search SDK: Merging data plane 2.0.6-preview and management 1.0.0-rc

### PR DESCRIPTION
**Pull request by bitsforthought**
FYI @chaosrealm @seansaleh

Merging all recent changes to both Search SDKs to the AutoRest branch. After this we will delete the search-v2-preview branch and create a new dev branch for data plane 3.x-rc.

### CODE REVIEW

**Comment by brjohnstmsft** [comment]
I'm a Microsoft FTE, so no CLA required.

File: src/Search/Search.Management.Tests/Tests/SearchServiceTests.cs [hunk 1]
```diff
@@ -184,6 +187,20 @@ public void CanScaleServiceUpAndDown()
...
+        public void CreateServiceWithInvalidNameGivesUsefulMessage()
+        {
+            Run(() =>
+            {
+                SearchManagementClient searchMgmt = GetSearchManagementClient();
+                SearchService service = DefineServiceWithSku(SkuName.Free);
+
+                CloudException e =
+                    Assert.Throws<CloudException>(() => searchMgmt.Services.CreateOrUpdate(Data.ResourceGroupName, InvalidServiceName, service));
+                Assert.Equal("Service name '----badname' is invalid: Service name must only contain lowercase letters, digits or dashes, cannot start or end with or contain consecutive dashes and is limited to 60 characters.", e.Message);
```
Comment (seansaleh) [inline src/Search/Search.Management.Tests/Tests/SearchServiceTests.cs]: You still have duplication of the InvalidServiceName, you could reference it in the assert here

Comment (brjohnstmsft) [inline src/Search/Search.Management.Tests/Tests/SearchServiceTests.cs]: Fixed

**Comment by markcowl** [comment]
@brjohnstmsft Can you look at the failure?

**Comment by brjohnstmsft** [comment]
@markcowl Looks like an infrastructure issue:

```
$ git fetch origin +refs/pull/2517/merge:
fatal: Couldn't find remote ref refs/pull/2517/merge
Unexpected end of command stream

The command "eval git fetch origin +refs/pull/2517/merge:" failed. Retrying, 2 of 3.

fatal: Couldn't find remote ref refs/pull/2517/merge
Unexpected end of command stream

The command "eval git fetch origin +refs/pull/2517/merge:" failed. Retrying, 3 of 3.

fatal: Couldn't find remote ref refs/pull/2517/merge
Unexpected end of command stream

The command "eval git fetch origin +refs/pull/2517/merge:" failed 3 times.



The command "git fetch origin +refs/pull/2517/merge:" failed and exited with 128 during .

Your build has been stopped.
```

I'll try closing and re-opening the PR.

**Comment by brjohnstmsft** [comment]
@markcowl Builds have passed.

**Pull request merged**